libs/wingsdk/src/std/map.ts

```diff
@@ -39,7 +39,7 @@ export class Map {
    * If the value that is associated to the provided key is an object, then you will get a reference
    * to that object and any change made to that object will effectively modify it inside the map.
    *
-   * @macro ($self$)[$args$]
+   * @macro ((obj, key) => { if (obj[key] === undefined) throw new Error(`Map does not contain key: "${key}"`); return obj[key]; })($self$, $args$)
```

Comment:
I want to raise a flag that there's secretly a bug here, because if a user has a `Map<str?>`, then calling `get` with a valid key could return `nil`, and since we represent empty optionals with JavaScript's `undefined`, this would throw an error.

I think the bug might be beyond the scope of this PR - could you open an issue about this?

Reply:
Interesting use case. One may argue that logically there is no difference between an optional without a value and no value at all.

As long as `has()` returns `false` in this case, I think this might be reasonable behavior.

All in all, out of scope (but let's open a tracking issue and see if this creates friction).

Reply:
Wild, yea great eye! Opened tracking issue: https://github.com/winglang/wing/issues/4525

Reply:
technically this should use `in`

```suggestion
   * @macro ((obj, key) => { if (!(key in obj)) throw new Error(`Map does not contain key: "${key}"`); return obj[key]; })($self$, $args$)
```

Reply:
> technically this should use `in`

🧠 yea that does make sense, it also squashes the bug. Incoming new commit